src/test/java/com/flightstats/hub/metrics/StatsdReporterIntegrationTest.java

```diff
@@ -43,26 +56,43 @@ private static StatsdReporter provideStatsDHandlers() {
         return provider.get();
     }
 
-    private static void runAfterSocketInit(Runnable runnable) {
-        Executors.newScheduledThreadPool(1).schedule(runnable, 100, TimeUnit.MILLISECONDS);
-    }
-
-
-    @BeforeClass
-    public static void startMockStatsDServer() {
+    public void startMockStatsDServer() throws InterruptedException, ExecutionException, TimeoutException {
+        CountDownLatch startupCountDownLatch = new CountDownLatch(2);
+        AtomicBoolean canListen = new AtomicBoolean();
         StatsdReporter handlers = provideStatsDHandlers();
-        IntegrationUdpServer udpServer = provideNewServer(metricsConfig.getStatsdPort());
-        IntegrationUdpServer udpServerDD = provideNewServer(metricsConfig.getDogstatsdPort());
-        runAfterSocketInit(() -> {
+        ExecutorService executorService = Executors.newFixedThreadPool(3);
+        IntegrationUdpServer udpServer = provideNewServer(metricsConfig.getStatsdPort(), startupCountDownLatch, canListen, executorService);
+        IntegrationUdpServer udpServerDD = provideNewServer(metricsConfig.getDogstatsdPort(), startupCountDownLatch, canListen, executorService);
+        CompletableFuture<String> sender = CompletableFuture.supplyAsync(() -> {
+            try {
+                startupCountDownLatch.await(15000, TimeUnit.MILLISECONDS);
```

Comment:
15 sec That's lot of time.  What exactly is this doing?  Even jetty server comes up in few seconds :)

Reply:
It's a lot because we don't lose that time if things return faster and having something outrageous seemed okish. We were having issues with 5s timeouts at some point in the lifetime of this test. I'll drop it to 5 and make sure things still pass consistently.